runtime/near-vm-logic/src/vmstate.rs

```diff
@@ -69,7 +69,6 @@ impl<'a> Memory<'a> {
         self.0.view_memory(slice).map_err(|_| HostError::MemoryAccessViolation.into())
     }
 
-    #[cfg(any(test, feature = "sandbox"))]
```

Comment:
No, I don’t see clean way in which it can be part of MemoryLike.  Mostly because it needs to account for gas which is something MemoryLike is normally completely unaware of.

I don’t think view_for_free is any more scary than any other operation for which we need to charge gas.  Not to look too far for examples, decoding UTF-16.